src/main/java/mate/academy/AsyncRequestProcessor.java

```diff
@@ -1,16 +1,24 @@
 package mate.academy;
 
+import java.util.Map;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.Executor;
 
 public class AsyncRequestProcessor {
     private final Executor executor;
+    private final Map<String, UserData> cache = new ConcurrentHashMap<>();
 
     public AsyncRequestProcessor(Executor executor) {
         this.executor = executor;
     }
 
     public CompletableFuture<UserData> processRequest(String userId) {
-        return null;
+        if (cache.containsKey(userId)) {
+            return CompletableFuture.supplyAsync(() -> cache.get(userId), executor);
+        } else {
+            cache.put(userId, new UserData(userId, "Details for " + userId));
```

Comment:
The creation of `UserData` and its addition to the cache should be done atomically to prevent concurrent access issues. You can use `ConcurrentHashMap.computeIfAbsent` to ensure that the value is computed and stored atomically.